packages/task/src/browser/task-service.ts

```diff
@@ -378,8 +403,57 @@ export class TaskService implements TaskConfigurationClient {
         this.logger.debug(`Task created. Task id: ${taskInfo.taskId}`);
 
         // open terminal widget if the task is based on a terminal process (type: shell)
+        // or attach to process output processor if a raw process (type: process)
         if (taskInfo.terminalId !== undefined) {
-            this.attach(taskInfo.terminalId, taskInfo.taskId);
+            if (taskInfo.config.type === 'shell') {
+                this.attach(taskInfo.terminalId, taskInfo.taskId);
+            } else {
+                this.attachProcess(taskInfo);
```

Comment:
In order to spawn processes and fetch the output in a different way than displaying it in a terminal widget, would creating a new task type solely for that be better? Or how should one hook onto one process but not the other, assuming that both `process` and `shell` tasks both are displayed in a terminal widget by default?